### ISSUE TITLE
Error: KV PUT failed: 429 Too Many Requests

### ISSUE THREAD

**Issue by dopufol** [rhinobase/hono-rate-limiter#34]
Workers KV has a maximum of 1 write to the same key per second. Writes made to the same key within 1 second will cause rate limiting (429) errors to be thrown.

To handle these errors, we recommend implementing a retry logic, with exponential backoff.

**Comment by MathurAditya724** [comment]
Yes, these are the limitations in the KV-based rate limiting, that's why I created support for the Durable Object Store too. The basic idea is that the KV is global and the Durable Object is regional, which helps it have an edge over KV.

**Comment by MathurAditya724** [comment]
Yes, this could be related to the issue. I think we should investigate whether the payload still expires if we `put` it without the `expiration` field. If it works as expected, we could remove the `expiration` from the `decrement` function entirely. This would simplify the payload to store only `totalHits`.

**Comment by MathurAditya724** [comment]
Related issue https://github.com/rhinobase/hono-rate-limiter/pull/40
